src/ServiceManagement/Profile/Commands.Profile/ServicePrincipal/GetAzureSubscriptionServicePrincipal.cs

```diff
@@ -0,0 +1,57 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.WindowsAzure.Management;
+using static Microsoft.WindowsAzure.Management.Models.SubscriptionServicePrincipalListResponse;
+
+namespace Microsoft.WindowsAzure.Commands.Profile
+{
+    [Cmdlet(VerbsCommon.Get, AzureSubscriptionServicePrincipalNounName, DefaultParameterSetName = ListAllServicePrincipalObjectIdsParameterSet)]
+    [OutputType(typeof(ServicePrincipal[]))]
```

Comment:
This should just be ServicePrincipal.  Also, you shoudl use a presentation type, so breakign changes in the cmdlet model type will not be automatically transferred to the cmdlet api surface